src/Core/CurrentContext.cs

```diff
@@ -148,6 +148,44 @@ public virtual Task SetContextAsync(ClaimsPrincipal user)
                         Type = OrganizationUserType.Manager
                     }));
             }
+
+            if (claimsDict.ContainsKey("orgcustom"))
+            {
+                Organizations.AddRange(claimsDict["orgcustom"].Select(c =>
+                    new CurrentContentOrganization
+                    {
+                        Id = new Guid(c.Value),
+                        Type = OrganizationUserType.Custom,
+                        AccessBusinessPortal = claimsDict.ContainsKey("accessbusinessportal") ? 
```

Comment:
This code smells like a new private method, anonymous method or extension method to me:

```c#
claimsDict.ContainsKey("accessbusinessportal") ? 
   claimsDict["accessbusinessportal"].Any(x => x.Value == c.Value) : false
```

Reply:
Implemented with this. 

```
private Permissions SetOrganizationPermissionsFromClaims(string organizationId, Dictionary<string, IEnumerable<Claim>> claimsDict)
{
    bool hasClaim(string claimKey) 
    {
        return claimsDict.ContainsKey(claimKey) ? 
            claimsDict[claimKey].Any(x => x.Value == organizationId) : false;
    }

    return new Permissions
    {
        AccessBusinessPortal = hasClaim("accessbusinessportal"),
        AccessEventLogs = hasClaim("accesseventlogs"),
        AccessImportExport = hasClaim(""),
        AccessReports = hasClaim("accessreports"),
        ManageAllCollections = hasClaim(""),
        ManageAssignedCollections = hasClaim(""),
        ManageGroups = hasClaim(""),
        ManagePolicies = hasClaim(""),
        ManageSso = hasClaim("managesso"),
        ManageUsers = hasClaim("manageusers")
    };
}
```